client/blocks/ecommerce-manage-nudge/index.js

```diff
@@ -0,0 +1,101 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal Dependencies
+ */
+import Card from 'components/card';
+import isECommerceManageNudgeDismissed from './selectors';
+import QueryPreferences from 'components/data/query-preferences';
+import { dismissNudge } from './actions';
+import { enhanceWithSiteType, recordTracksEvent } from 'state/analytics/actions';
+import { withEnhancers } from 'state/utils';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class ECommerceManageNudge extends Component {
+	static propTypes = {
+		isDismissed: PropTypes.bool.isRequired,
+		recordTracksEvent: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	componentDidMount() {
+		this.recordView();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.siteId && this.props.siteId && this.props.siteId !== prevProps.siteId ) {
+			this.recordView();
+		}
+	}
+
+	recordView() {
+		if ( ! this.props.isDismissed ) {
+			this.props.recordTracksEvent( 'calypso_ecommerce_manage_stats_nudge_view' );
+		}
+	}
+
+	onDismissClick = () => {
+		this.props.recordTracksEvent( 'calypso_ecommerce_manage_stats_nudge_dismiss_icon_click' );
+		this.props.dismissNudge();
+	};
+
+	render() {
+		const { translate } = this.props;
+
+		if ( this.props.isDismissed ) {
```

Comment:
Sorry I missed this the first go 'round but do we still need to verify `siteId` is present and return null here too? I see we have it as a required prop, but not sure if we still need to check for it all the time in calypso code ( been a long while since i've written code here now )

Reply:
Cleared out my localStorage and all seems fine if siteId is not present so disregard :)